$OpenBSD: patch-configure_ac,v 1.18 2018/07/11 12:03:22 ajacoutot Exp $

XXX make sure gint64 matches int64_t so that C++ is happy
https://gitlab.gnome.org/GNOME/glib/issues/1442

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -3106,13 +3106,26 @@ $ac_cv_sizeof_int)
   guint64_constant='(val)'
   ;;
 $ac_cv_sizeof_long)
-  gint64=long
-  gint64_modifier='"l"'
-  gint64_format='"li"'
-  guint64_format='"lu"'
-  glib_extension=
-  gint64_constant='(val##L)'
-  guint64_constant='(val##UL)'
+  case $host in
+  *-*-openbsd*)
+    gint64='long long'
+    gint64_modifier='"ll"'
+    gint64_format='"lli"'
+    guint64_format='"llu"'
+    glib_extension=
+    gint64_constant='(val##LL)'
+    guint64_constant='(val##ULL)'
+    ;;
+  *)
+    gint64=long
+    gint64_modifier='"l"'
+    gint64_format='"li"'
+    guint64_format='"lu"'
+    glib_extension=
+    gint64_constant='(val##L)'
+    guint64_constant='(val##UL)'
+    ;;
+  esac
   ;;
 $ac_cv_sizeof_long_long)
   gint64='long long'
@@ -3349,7 +3362,14 @@ $ac_cv_sizeof_int)
   gint64=int
   ;;
 $ac_cv_sizeof_long)
-  gint64=long
+  case $host in
+  *-*-openbsd*)
+    gint64='long long'
+    ;;
+  *)
+    gint64=long
+    ;;
+  esac
   ;;
 $ac_cv_sizeof_long_long)
   gint64='long long'
